### ISSUE TITLE
revision d2697a9 breaks logging into gnome ($USER logic)

### ISSUE THREAD

**Issue by ht990332** [systemd/systemd#5246]
### Submission type

  - [x] Bug report
  - [ ] Request for enhancement (RFE)

With revision d2697a9, logging into gnome is broken (I have autologin enabled).
Downgrading one revision behind to revision d20a328f9 fixes everything.
I am attaching journalctl -b output.
[badlog.txt](https://github.com/systemd/systemd/files/755243/badlog.txt)

Notice the part:
eb 06 17:19:31 hades systemd[542]: Failed to subscribe to activation signal: Connection timed out
Feb 06 17:19:31 hades systemd[542]: Failed to register name: Connection timed out
Feb 06 17:19:31 hades systemd[542]: Failed to set up API bus: Connection timed out
Feb 06 17:19:31 hades systemd[542]: dbus.service: Main process exited, code=exited, status=216/GROUP
Feb 06 17:19:31 hades systemd[542]: dbus.service: Unit entered failed state.
Feb 06 17:19:31 hades systemd[542]: dbus.service: Failed with result 'exit-code'.
Feb 06 17:19:31 hades systemd[542]: Looping too fast. Throttling execution a little.
Feb 06 17:19:32 hades systemd[542]: Looping too fast. Throttling execution a little.
Feb 06 17:19:33 hades systemd[542]: Looping too fast. Throttling execution a little.
Feb 06 17:19:34 hades systemd[542]: Looping too fast. Throttling execution a little.
Feb 06 17:19:35 hades systemd[542]: Looping too fast. Throttling execution a little.

**Comment by ht990332** [comment]
Is there any way I can get more verbose details on what exactly is happening?
I know for sure which last revision works and which first revision is broken.

**Comment by evverx** [comment]
Also, https://github.com/systemd/systemd/commit/d2697a95fede9d51f2cd9a6c9184cd3971973478 breaks https://github.com/systemd/systemd-centos-ci/blob/master/slave/tests/loginctl/runtest.sh#L125 somehow

```
:: [  BEGIN   ] :: Running 'userdel -r systemdtester'
userdel: user systemdtester is currently used by process 4247
:: [   FAIL   ] :: Command 'userdel -r systemdtester' (Expected 0, got 8)
```

**Comment by gdamjan** [comment]
I can confirm the issue without Gnome/GDM. I've done tests via ssh and terminal login.

The original issue is that dbus didn't start, and from vaguelly reading thourgh the code and the patch it seems to assume that a service without User= means User=root. Well that's not true for --user services, which don't have an User= but it means the current user.

Basically, all --user services are broken. `busctl --user` doesn't work, `systemctl --user` doesn't work either.

**Comment by gdamjan** [comment]
ps. adding `Environment=SYSTEMD_LOG_LEVEL=debug` to `user@1000.service` and this is part of the log:

```
Feb 06 22:08:22 arch-uefi-test systemd[537]: Skipping PR_SET_MM_ARG_START, as we don't have privileges.
Feb 06 22:08:22 arch-uefi-test systemd[537]: dbus.socket: Failed at step GROUP spawning /usr/bin/systemctl: Operation not permitted
Feb 06 22:08:22 arch-uefi-test systemd[533]: dbus.socket: User lookup succeeded: uid=0 gid=0
Feb 06 22:08:22 arch-uefi-test systemd[533]: Received SIGCHLD from PID 537 ((ystemctl)).
Feb 06 22:08:22 arch-uefi-test systemd[533]: Child 537 ((ystemctl)) died (code=exited, status=216/GROUP)
Feb 06 22:08:22 arch-uefi-test systemd[533]: dbus.socket: Child 537 belongs to dbus.socket
Feb 06 22:08:22 arch-uefi-test systemd[533]: dbus.socket: Control process exited, code=exited status=216
Feb 06 22:08:22 arch-uefi-test systemd[533]: dbus.socket: Got final SIGCHLD for state start-post
Feb 06 22:08:22 arch-uefi-test systemd[533]: dbus.socket: Changed start-post -> listening
Feb 06 22:08:22 arch-uefi-test systemd[533]: dbus.socket: Job dbus.socket/start finished, result=done
Feb 06 22:08:22 arch-uefi-test systemd[533]: Listening on D-Bus User Message Bus Socket.
```

**Comment by poettering** [comment]
the offending patch simply ignore "--user" mode entirely, and harcoded "root" as the default user if none was specified. That's the right thing to do for --system, but certainly not for --user...

**Comment by poettering** [comment]
Fix pending in #5287, please have a look.
